esdt-safe/src/from_sovereign/transfer_tokens.rs

```diff
@@ -56,41 +55,39 @@ pub trait TransferTokensModule:
         let mut output_payments = ManagedVec::new();
 
         for operation_token in operation_tokens.iter() {
-            let mx_token_id_state = self
-                .sovereign_to_multiversx_token_id(&operation_token.token_identifier)
-                .get();
-
-            let mx_token_id = match mx_token_id_state {
-                // token is from sovereign -> continue and mint
-                TokenMapperState::Token(token_id) => token_id,
-                // token is from mainchain -> push token
-                _ => {
-                    output_payments.push(operation_token.clone());
-
-                    continue;
-                }
-            };
+            let sov_to_mvx_token_id_mapper =
+                self.sovereign_to_multiversx_token_id_mapper(&operation_token.token_identifier);
+
+            // token is from mainchain -> push token
+            if sov_to_mvx_token_id_mapper.is_empty() {
+                output_payments.push(operation_token.clone());
+
+                continue;
+            }
+
+            // token is from sovereign -> continue and mint
+            let mvx_token_id = sov_to_mvx_token_id_mapper.get();
 
             if operation_token.token_nonce == 0 {
                 self.tx()
                     .to(ToSelf)
                     .typed(system_proxy::UserBuiltinProxy)
-                    .esdt_local_mint(&mx_token_id, 0, &operation_token.token_data.amount)
+                    .esdt_local_mint(&mvx_token_id, 0, &operation_token.token_data.amount)
                     .transfer_execute();
 
                 output_payments.push(OperationEsdtPayment {
-                    token_identifier: mx_token_id,
+                    token_identifier: mvx_token_id,
                     token_nonce: 0,
                     token_data: operation_token.token_data,
                 });
 
                 continue;
             }
 
-            let nft_nonce = self.mint_and_save_token(&mx_token_id, &operation_token);
+            let nft_nonce = self.mint_and_save_token(&mvx_token_id, &operation_token);
```

Comment:
```suggestion
            let mvx_nft_nonce = self.mint_and_save_token(&mvx_token_id, &operation_token);
```
-----
Also, let's rename func `mint_and_save_token` to `esdt_create_and_update_mapper`

Reply:
I also added another function to update the mappers